src/components/global-toolbar/global-toolbar.test.js

```diff
@@ -0,0 +1,90 @@
+import React from 'react';
+import ConnectedGlobalToolbar, {
+  GlobalToolbar,
+  mapStateToProps,
+  mapDispatchToProps,
+} from './index';
+import { mockState, setup } from '../../utils/state.mock';
+
+describe('GlobalToolbar', () => {
+  it('renders without crashing', () => {
+    const wrapper = setup.mount(<ConnectedGlobalToolbar />);
+    expect(wrapper.find('.pipeline-icon-toolbar__button').length).toBe(5);
+  });
+
+  it('hides all buttons (except logo and route buttons) when visible prop is false for each of them', () => {
+    const visible = {
+      themeBtn: false,
+      settingsBtn: false,
+    };
+    const wrapper = setup.mount(<ConnectedGlobalToolbar />, { visible });
+    expect(wrapper.find('.pipeline-icon-toolbar__button').length).toBe(3);
+  });
+
+  it('hides one button when visible prop is false for one of them', () => {
+    const visible = {
+      themeBtn: false,
+    };
+    const wrapper = setup.mount(<ConnectedGlobalToolbar />, { visible });
+    expect(wrapper.find('.pipeline-icon-toolbar__button').length).toBe(4);
+  });
```

Comment:
Looking at these tests, it does remind me that there is likely no need for setting the `visible` prop for the theme and settings panel button - there simply is no cases at all within our codebase for `visible.themeBtn` or `visible.settingsBtn` to be false within our codebase ( or in general anywhere in the app.) 

This just reminds me that we would need to revisit the current set of state items that existed - for now I would say that the `visible` prop for the settings panel and theme button are not needed given the lack of a use case for it, and we can delete these tests to help this component and set of tests to be lighter weight.  

( You definitely did the right thing here as I completely understand that you had followed how we had previously set up the theme button and settings panel, and definitely nothing wrong with these tests here. )

Reply:
Removed!